### PR TITLE
JENKINS-34262 OAuth implemented, building, skipping tests

### PR DESCRIPTION
**Pull request by Strangerxxx**
It is fast and dirty implementation of OAuth credentials authentication, which builds and works, but for faster build i skipped tests.
It's able to resolve [JENKINS-34262](https://issues.jenkins-ci.org/browse/JENKINS-34262)

### CODE REVIEW

File: src/main/java/com/cloudbees/jenkins/plugins/bitbucket/client/BitbucketCloudApiClient.java [hunk 1]
```diff
@@ -23,17 +23,7 @@
...
-import com.cloudbees.jenkins.plugins.bitbucket.api.BitbucketCommit;
-import com.cloudbees.jenkins.plugins.bitbucket.api.BitbucketException;
-import com.cloudbees.jenkins.plugins.bitbucket.api.BitbucketPullRequest;
-import com.cloudbees.jenkins.plugins.bitbucket.api.BitbucketRepository;
-import com.cloudbees.jenkins.plugins.bitbucket.api.BitbucketRepositoryProtocol;
-import com.cloudbees.jenkins.plugins.bitbucket.api.BitbucketRepositoryType;
-import com.cloudbees.jenkins.plugins.bitbucket.api.BitbucketRequestException;
-import com.cloudbees.jenkins.plugins.bitbucket.api.BitbucketTeam;
-import com.cloudbees.jenkins.plugins.bitbucket.api.BitbucketWebHook;
+import com.cloudbees.jenkins.plugins.bitbucket.api.*;
```
Comment (casz) [inline src/main/java/com/cloudbees/jenkins/plugins/bitbucket/client/BitbucketCloudApiClient.java]: Explicit imports, please 👮

**Comment by kwin** [comment]
This only supports OAuth for Bitbucket Cloud but not for Bitbucket Server. Is that an oversight?

**Comment by Strangerxxx** [comment]
@kwin seems like. Im just using Cloud, and no need in Server support. Don't know how it would be on server.

**Comment by Strangerxxx** [comment]
Tests won't pass :confused:

**Comment by casz** [comment]
```
[INFO] Found reliance on default encoding in 
com.cloudbees.jenkins.plugins.bitbucket.api.BitbucketOAuthService.getHttpBasicAuthHeaderValue(): 

String.getBytes() 
[com.cloudbees.jenkins.plugins.bitbucket.api.BitbucketOAuthService] At BitbucketOAuthService.java:[line 39] DM_DEFAULT_ENCODING
```
https://ci.jenkins.io/blue/organizations/jenkins/Plugins%2Fbitbucket-branch-source-plugin/detail/PR-69/4/pipeline/8#step-30-log-280

**Comment by casz** [comment]
Need to set default encoding usually UTF-8

**Comment by casz** [comment]
One test failing
https://ci.jenkins.io/blue/organizations/jenkins/Plugins%2Fbitbucket-branch-source-plugin/detail/PR-69/5/tests

**Comment by casz** [comment]
Seems unrelated to your changes

**Comment by Strangerxxx** [comment]
Related.. caused by using base64. Fixed by last commit

File: src/main/java/com/cloudbees/jenkins/plugins/bitbucket/api/BitbucketOAuthService.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
+package com.cloudbees.jenkins.plugins.bitbucket.api;
+
+import org.eclipse.jgit.util.Base64;
```
Comment (kwin) [inline src/main/java/com/cloudbees/jenkins/plugins/bitbucket/api/BitbucketOAuthService.java]: This is not base64 from commons codec.

Comment (Strangerxxx) [reply src/main/java/com/cloudbees/jenkins/plugins/bitbucket/api/BitbucketOAuthService.java]: yes, but it uses base32 from commons
btw, test passed

**Comment by kwin** [comment]
Now bitbucket server supports personal access tokens: https://confluence.atlassian.com/bitbucketserver/bitbucket-server-5-5-release-notes-938037662.html. Would it be possible to support that as well with this PR?

**Comment by Strangerxxx** [comment]
@kwin, I have no bitbucket server installation, so cannot test it out. I wouldn't implement something which i couldn't test.

File: src/main/java/com/cloudbees/jenkins/plugins/bitbucket/client/BitbucketCloudApiClient.java [hunk 3]
```diff
@@ -504,6 +518,37 @@ public BitbucketTeam getTeam() throws IOException, InterruptedException {
         return getRepositories(null);
     }
 
+    private synchronized HttpClient getHttpClient() {
+        if (this.client == null) {
+            HttpClient client = new HttpClient(connectionManager);
```
Comment (tomfitzhenry) [inline src/main/java/com/cloudbees/jenkins/plugins/bitbucket/client/BitbucketCloudApiClient.java]: Why does a new HttpClient need to be instantiated? There's already one instantiated in the constructor of BitbucketCloudApiClient.

**Comment by casz** [comment]
Should be reworked after #101 is merged

**Comment by casz** [comment]
Please consider reworking the PR or work together with #68 now that #101 is merged.

**Comment by danielfbm** [comment]
@Strangerxxx do you need any help with this PR? we are kind of in need of this update and are willing to contribute in anyway possible

**Pull request closed without merging**